Hai secret no more (#4952)
endless-sky/endless-sky: data/hai/hai reveal 3 early.txt

# Copyright (c) 2020 MasterOfGrey
#
# Endless Sky is free software: you can redistribute it and/or modify it under the
# terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later version.
#
# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
# PARTICULAR PURPOSE.  See the GNU General Public License for more details.

########################################################################

event "secret mountaintop"
	system "Ultima Thule"

> 		object "Wormhole Alpha"

Author: The wormhole alpha government is already included in the object under the planet listing, this is just a modification to the existing object by restating its stats and adding the orbiting station.

> 		object "Wormhole Alpha"

Reviewer: You're right. It is unneeded. I'll test to make sure, but I'm 98.74% certain that you are correct.

			`	"I think we've generally had enough of this here. Answer me two questions, and then we shall make our decision."`
			`	"Do you think that a diplomatic post should be able to defend itself? Or should it trust to the strength of those who meet there to ensure its protection?"`
			`	This seems like a very strange question...`
			choice
				`	"It should be strong enough to defend itself. You cannot trust your enemy."`
					goto trust
				`	"Both sides must be trusted to protect a diplomatic post, or there is no point to having one."`
					goto caution
			label trust
			apply
				set haitrust
			`	You seem to get no reaction, and the Elder follows up with, "What is the gravest threat, in your opinion, facing humanity?"`
				goto return
			label caution
			apply
				set haicaution
			`	You seem to get no reaction, and the Elder follows up with, "What is the gravest threat, in your opinion, facing humanity?"`
			label return
			choice
				`	"The Pug."`
				`	"The Alphas."`
				`	"Unknown Jump-Capable Aliens."`
				`	"Experimental Technologies."`

> 				`	"Hubris."`

Author: ```suggestion
				`	"Humanity's Hubris."`
```

			object Mountaintop
				sprite "planet/station hai eight geocoris dim"
				distance 315
				period 31.1964
	planet Mountaintop
		government "Hai"
		attributes uninhabited
		landscape land/station0
		description `This station is made of eight Geocoris ships dismantled and reassembled into a spherical structure with external storage and landing platforms. Hai workers are shuffling about the station, installing human amenities and vast holographic projection systems. Presently it looks much like a construction site, but it is rapidly approaching completion.`
		security 0.2

mission "Hai Building Mountaintop [2] Recruits"
	name "Encourage recruits for Mountaintop"
	description "Now that the station is going ahead, the Hai need to recruit people to play the part. Someone needs to sell it to them, and apparently that someone is you."
	landing
	source "Hai-home"
	stopover "Makerplace"
	stopover "Darkmetal"
	stopover "Giverstone"
	stopover "Stonebreak"
	stopover "Cloudfire"
	stopover "Snowfeather"
	stopover "Icelake"
	stopover "Newhome"
	stopover "Frostmark"
	stopover "Heartvalley"
	stopover "Allhome"
	stopover "Dustmaker"
	stopover "Greenbloom"
	stopover "Skyfarm"
	stopover "Farwater"

> 	stopover "Greenwater"

Author: This one is meant to make you slightly resentful of it. The Narratives in ES, while covering vastly different scenarios, have very similar themes. In this case you've been dragged into a diplomatic mess, albeit one with a slightly non-serious bent. It will encourage you to appreciate it more when you go back into doing other major quests with more seriousness but less dumb politics afterwards. :P

> 	stopover "Greenwater"

Reviewer 2: To expand on this slightly, presumably the same models Ms Adrianopoulos (I imagine she'd have my bank accounts frozen if I presumed to call her Terry without an invitation) uses to determine the number of people needed on the station would also provide some insight into where recruiting could be most effective.

> 	stopover "Greenwater"

Reviewer 2: Or, if you want to be less subtle about it, have `ZZ Hai Building Mountaintop [2a] Sales Job` keep track of how many times you've done this and get grumpier as you go.

`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. The routine is well-practiced now, but starting to get a little tiresome.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You struggle to maintain a cheerful and enthusiastic facade as you count down the minutes until this ordeal is over.`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`

(That last one is for when you actually *have* done it more times than there are habitable planets in Hai space. The mission fires every time you land, even if you've been there before.)

> 	stopover "Greenwater"

Author: Ok, if i can work out how to do it _that_ sounds like an approach I will take.

> 	stopover "Greenwater"

Reviewer 2: Would something like this work?
```
	on offer
		"hai dog and pony" ++
		conversation
			branch toomany
				"hai dog and pony" < 17
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				decline
			label toomany
			branch interminable
				"hai dog and pony" < 14
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				decline
			label interminable
			[...]
			label tiresome
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
				decline
```

I haven't tested it, but I'm pretty sure all the individual pieces work. The slightly odd structure lets us keep each part together as a self-contained unit, but the following should be equivalent and might be easier to understand:

```
	on offer
		"hai dog and pony" ++
		conversation
			branch toomany
				"hai dog and pony" >= 17
			branch interminable
				"hai dog and pony" >= 14
			[...]
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
				decline
			label tiresome
			[...]
			label interminable
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				decline
			label toomany
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				decline
```

Note that in both cases we increment the counter before the conversation, so it will always include the current mission.

> 	stopover "Greenwater"

Author: Oh I like this, I believe one of them will work. I'll see how I go getting it in to work. Do you have a snapshot from just before this that you might be able to use to test for me by any chance?  (or give me to to use to test)

> 	stopover "Greenwater"

Author: Long delay I know, but I've now implemented this suggestion I believe.

			choice
				`	"Is it me?"`
			`Sayari winks at you, "Catch on quick don't you."`
			`	"The actions you're recorded as having taken show a certain conscience for the good of your species as a whole, despite your affiliations. I have convinced the other diplomats and the Elders that you might be a viable solution to this impasse. The human diplomats have already agreed, but the Elders wish to speak with you first."`
			choice
				`	"Who are these Elders?"`
				`	"Of course, that's natural."`
					goto main
			`	Sayari explains, "The Hai are ruled by an elected 'Council of Elders', they are not responsible for every administrative decision that occurs but they are the highest authority of our government. Their decisions are final, at least for the term of an election."`
			label main
			`	"If you're willing, there will be a council meeting in an hour, I can put you first on the agenda. Will you do this?" asks Sayari.`
			choice
				`	"I suppose I must."`
					goto council
				`	"Someone better be paying me for this."`
					goto council
				`	"Of course, let's do it."`
					goto council
				`	"Maybe in a bit, I should do something else first."`
					defer
			label council

> 			`Following Sayari you pass beyond a security checkpoint where Xilin is prevented from following you. Sayari speaks with another Hai doing administrative work for a moment before gesturing you to a private antechamber.`

Reviewer 3: Might be better said as 'gesturing you towards a private antechamber' or 'directing you to a private antechamber'

			`	You seem to get no reaction and the Elder follows up with, "What is the gravest threat, in your opinion, facing humanity?"`
			label return
			choice
				`	"The Pug"`
				`	"The Alphas"`
				`	"Unknown Jump-Capable Aliens"`
				`	"Experimental Technologies"`
				`	"Hubris"`
					goto wise
			`	There is a collection of chittering laughs from the various council members. You're not sure if there was something wrong with what you said but you aren't left long to ponder.`
			`	"It seems the human agent is an honest fool, but an honest fool can at least be trusted to be foolish honestly on all fronts."`
			`	You're pretty sure you should be offended by that, but the councilmembers are smiling and don't look like they're about to evict you.`
				goto end
			label wise
			`	"It seems you learnt the correct lesson from the crisis of your species," he says.`
			`	You see several appreciative nods happening around the room out of the corner of your eyes.`
			label end
			`	"You may act as the oversight for the human governments in this project. Proposal D-86513-7P4 is approved. Council adjourned."`
			`	When you meet Sayari afterwards a touch of her usual enthusiasm has returned.`

> 			`	"This is fantastic news, now all that remains to be done is for you to escort the construction fleet from Makerplace to Ultima Thule and observe the initial stages. They'll be waiting for your arrival on Makerplace."`

Reviewer 3: Just a small continuity thing, Sayari wouldn't know how the council meeting went as she was in another room, maybe this could be fixed by saying 'When you meet back with Sayari in the antechamber and tell her the good news, her face lights up with enthusiasm'

				accept
	on accept
		event "role-play station before construction"

> 		log `The diplomats have elected to double-down on the role-play community angle, citing it's success. However, the Hai are unwilling to trust a human government to see the inner workings of a station under construction in the wake of the civil war, and the Republic will not allow a station to be built on their side unsupervised. You have been asked to fulfill the role of supervisor as a result. The Elders asked you some odd questions, but agreed that you were an acceptable individual for the task.`

Reviewer 3: misuse of its

			`	"Of course you do, everyone likes to make money," she says.`
			`	"The Hai have put out the call for volunteers to staff the new Mountaintop station when it's finished. While you've been off watching early construction works, we've been gauging public interest and well... it's not going to work. In order to sell this they're going to need at least four thousand volunteers by the end of the week to get a good handle on what they can expect to need in terms of facilities, amenities, supplies, and various livability requirements. Nearly twelve thousand by the time the station is actually built. Our current estimates suggest they'll reach a quarter of that."`
			`	As she's talking Terry has been laying out charts and graphs, and using a handheld Hai holographic device that she seems to have picked up somewhere to show animated diagrams.`
			choice
				`	"How do you know all this?"`
			`	Terry raises an eyebrow at you, "You remember we have a supercomputer that could predict the economic situation across the whole of human habited space up to fifty years in the future to within 1% accuracy right?"`

> 			`	When she puts it that way, it does suddenly feel like a bit of a stupid question.`

Reviewer 3: Just a thought, the pc could answer back with 'and how well did that work out for you?' if you want some branching dialogue here.

		object
			sprite planet/europa-b
			distance 403.41
			period 64.056
		object
			sprite planet/desert7-b
			distance 824.9
			period 187.302
		object
			sprite planet/gas1
			distance 1706.99
			period 557.553
			object
				sprite planet/ice8-b
				distance 216
				period 11.1964
		object "Wormhole Alpha"
			sprite planet/wormhole
			distance 4208.63
			period 2158.5
			object Mountaintop
				sprite "effect/flotsam yottrite"
				distance 315
				period 31.1964
	planet Mountaintop
		government "Unknown"
		# Placeholder description; you can't actually land.
		description `This tiny asteroid has a space hazard warning transmitter, of a type typically used to mark construction sites. Perhaps someone intends to build a space station here?`
		"required reputation" 1000
		bribe 0

mission "Hai Building Mountaintop [1] Construction Fleet"
	name "Escort construction fleet to Ultima Thule"
	description "Meet a construction fleet at <stopovers> and escort them to to <destination> to prevent a confrontation."
	autosave
	landing
	source "Hai-home"
	stopover "Makerplace"
	destination "Mountaintop"
	clearance
	to offer
		has "Hai Leaks Response 6: done"
	on offer
		conversation

> 			`When you return to Hai-home you are directed now to a government hangar where you are met by Xilin and Sayari, both looking very relieved.`

Author: These commas make it feel stilted to me, you sure?

> 			`When you return to Hai-home you are directed now to a government hangar where you are met by Xilin and Sayari, both looking very relieved.`

Reviewer 4: Just double checked both of them. The first is definitely necessary, but the second is up for debate since it's hard to determine whether it's a restrictive clause, so I think taking it out wouldn't be a huge deal.
```suggestion
			`When you return to Hai-home, you are directed to a government hangar where you are met by Xilin and Sayari, both looking very relieved.`
```
```

> 			`	"I am very glad to see you returned Captain <last>, Sayari said you were away on an important errand but I've been hoping for a day now you would appear, as I think you might be the solution to the current impasse."`

Author: You seem to have reviewed the old version of a line that was edited already a few days ago here.

> 			`	"You see," says Sayari, "in the light of the recent civil war your species has had, some of the Elder's are less than keen with allowing a partisan observer to see the inner workings of a station in its earliest phases, especially when we haven't built one in so, so long."`

Author: ```suggestion
			`	"You see," says Sayari, "in the light of the recent civil war your species has had, some of the Elders are less than keen on allowing a partisan observer to see the inner workings of a station in its earliest phases, especially when we haven't built one in so, so long."`
```
Not keen on throwing out especially. I think it changes the tone too much.

> 			`	"You see," says Sayari, "in the light of the recent civil war your species has had, some of the Elder's are less than keen with allowing a partisan observer to see the inner workings of a station in its earliest phases, especially when we haven't built one in so, so long."`

Author: Another old version.

			`	Now that she mentions it, you don't think you've seen a Hai station at all.`
			`	"However, therein lies a problem," says Xilin.`
			`	"The Republic will not allow the Hai to build a station this side of the wormhole without some oversight by someone we trust. This has now been a point of contention for some days, and multiple suggestions have been rejected."`
			`	Sayari chitters, "Honestly it's all very silly. Lots of old men who are meant to be friends concerned that the other might have been lying to them."`
			`	In an oddly comical gesture they both shrug at the same time.`

> 			`	"Anyway, thanks to a little Freedom of Information request to the Republic, I have been looking at things and I think I have the solution."`

Author: `They make occasional grammatical errors, particularly leaving out articles or inserting the where they are not needed.` -The Hai
I like Sayari's speech to have the occasional, subtle inefficiency in it.

> 			`	"Anyway, thanks to a little Freedom of Information request to the Republic, I have been looking at things and I think I have the solution."`

Author: ```suggestion
			`	"Anyway, thanks to a little Freedom of Information request to the Republic, I have been looking at things, and I think I have the solution."`
```

			choice
				`	"Is it me?"`

> 			`Sayari winks at you, "Catch on quick don't you."`

Author: Another old line

					goto main
			`	Sayari explains, "The Hai are ruled by an elected council of elders. They are not responsible for every administrative decision that occurs, but they are the highest authority of our government. Their decisions are final, at least for the term of an election."`
			label main
			`	"If you're willing, there will be a council meeting in an hour, I can put you first on the agenda. Will you do this?" asks Sayari.`
			choice
				`	"I suppose I must."`
					goto council
				`	"Someone better be paying me for this."`
					goto council
				`	"Of course, let's do it."`
					goto council
				`	"Maybe in a bit, I should do something else first."`
					defer
			label council
			`Following Sayari you pass beyond a security checkpoint where Xilin is prevented from following you. Sayari speaks with another Hai doing administrative work for a moment before gesturing you towards a private antechamber.`
			`	"We shouldn't have to wait very long, but while we're alone," she glances at the door as if faintly nervous for a second, "did you find anything on your investigation?"`
			choice
				`	"Yes, actually I found proof."`
				`	"Unfortunately, I did find something."`

> 			`	You hand Sayari a data stick. "I have saved everything from my logs on there."`

Author: See prev on Sayari's speech.

			`	Sayari looks crestfallen and takes the data stick gingerly, looking at it as if it were some snake that might bite her at any moment.`
			`	"I will have a look," is all she says.`
			`	No sooner has she pocketed it than there is a knock at the door and an aide looks in, "The Council will see you now."`

> 			`	The next twenty minutes prove to be an exercise in tedium as various members of the council interrogate you over apparently mundane details of your recent actions as recorded in the FOI report. You're not really sure what most of it has to do with anything, but eventually one of the elders stands up.`

Author: Ah yes, this isn't speech so it should be in full form. Good catch.

> 			`	The next twenty minutes prove to be an exercise in tedium as various members of the council interrogate you over apparently mundane details of your recent actions as recorded in the FOI report. You're not really sure what most of it has to do with anything, but eventually one of the elders stands up.`

Author: ```suggestion
			`	The next twenty minutes prove to be an exercise in tedium as various members of the council interrogate you over apparently mundane details of your recent actions as recorded in the Freedom of Information report. You're not really sure what most of it has to do with anything, but eventually one of the elders stands up.`
```

			`	"Do you think that a diplomatic post should be able to defend itself? Or should it trust to the strength of those who meet there to ensure its protection?"`
			`	This seems like a very strange question...`
			choice
				`	"It should be strong enough to defend itself. You cannot trust your enemy."`
					goto trust
				`	"Both sides must be trusted to protect a diplomatic post, or there is no point to having one."`
					goto caution
			label trust
			apply
				set haitrust
			`	You seem to get no reaction and the Elder follows up with, "What is the gravest threat, in your opinion, facing humanity?"`
				goto return
			label caution
			apply
				set haicaution
			`	You seem to get no reaction and the Elder follows up with, "What is the gravest threat, in your opinion, facing humanity?"`
			label return
			choice
				`	"The Pug"`
				`	"The Alphas"`
				`	"Unknown Jump-Capable Aliens"`
				`	"Experimental Technologies"`
				`	"Hubris"`
					goto wise
			`	There is a collection of chittering laughs from the various council members. You're not sure if there was something wrong with what you said but you aren't left long to ponder.`

> 			`	"It seems the human agent is an honest fool, but an honest fool can at least be trusted to be foolish honestly on all fronts."`

Author: I have a rough idea. It may be explored elsewhere in the future.

			`	You're pretty sure you should be offended by that, but the council members are smiling and don't look like they're about to evict you.`
				goto end
			label wise
			`	"It seems you learnt the correct lesson from the crisis of your species," he says.`
			`	You see several appreciative nods happening around the room out of the corner of your eyes.`
			label end
			`	"You may act as the oversight for the human governments in this project. Proposal D-86513-7P4 is approved. Council adjourned."`

> 			`	When you meet Sayari afterwards a touch of her usual enthusiasm returns as you share the news.`

Reviewer 4: ```suggestion
			`	When you meet Sayari afterwards, a touch of her usual enthusiasm returns as you share the news.`
```

> 			`	"This is fantastic news, now all that remains to be done is for you to escort the construction fleet from Makerplace to Ultima Thule and observe the initial stages. They'll be waiting for your arrival on Makerplace."`

Reviewer 4: ```suggestion
			`	"This is fantastic news. Now all that remains to be done is for you to escort the construction fleet from <stopovers> to Ultima Thule and observe the initial stages. They'll be waiting for your arrival."`
```
Saying the stopover twice seems a little redundant is all. You could also do an exclamation point if you think it better fits Sayari's mood, but there probably needs to be a pause longer than a comma there.

				accept
	on accept
		event "role-play station before construction"

> 		log `The diplomats have elected to double-down on the role-play community angle, citing its success. However, the Hai are unwilling to trust a human government to see the inner workings of a station under construction in the wake of the civil war, and the Republic will not allow a station to be built on their side unsupervised. You have been asked to fulfill the role of supervisor as a result. The Elders asked you some odd questions, but agreed that you were an acceptable individual for the task.`

Reviewer 4: ```suggestion
		log `The diplomats have elected to double-down on the role-play community angle, citing its success. The Hai, however, are unwilling to trust a human government to see the inner workings of a station under construction in the wake of the civil war, and the Republic will not allow a station to be built on their side unsupervised. You have been asked to fulfill the role of supervisor as a result. The Elders asked you some odd questions, but agreed that you were an acceptable individual for the task.`
```
I learned as a stylistic rule to never start a sentence with "however" if you can avoid it, but up to you.

Author: Should be a ; before "however" then.

Reviewer 4: The phrase has to be a complete sentence before a semicolon. I think there's an exception to the rule of thumb where if the `however` at the start feels absolutely necessary, you use a semicolon instead of a period at the end of the previous sentence. Are you thinking of that? In that case, it'd look like this:
```suggestion
		log `The diplomats have elected to double-down on the role-play community angle, citing its success; however, the Hai are unwilling to trust a human government to see the inner workings of a station under construction in the wake of the civil war, and the Republic will not allow a station to be built on their side unsupervised. You have been asked to fulfill the role of supervisor as a result. The Elders asked you some odd questions, but agreed that you were an acceptable individual for the task.`
```

	on complete
		event "role-play station after construction"
	

mission "Hai Building Mountaintop [1a] Construction Escorts"
	landing
	invisible
	source "Makerplace"
	destination "Mountaintop"
	clearance
	to offer
		has "Hai Building Mountaintop [1] Construction Fleet: active"
	on visit
		dialog `You have arrived without the seven Geocoris you are escorting from <origin> to <destination>. You must take off and wait for them to finish this mission.`
	npc accompany save
		personality escort timid
		government "Hai (Wormhole Access)"
		ship Geocoris "Du Cor Hey"
		ship Geocoris "Mon Yo Sen"
		ship Geocoris "Vai e Lur"
		ship Geocoris "Du Woa Ow"
		ship Geocoris "Luf Sook Hey"
		ship Geocoris "Du Eh"
		ship Geocoris "Mon e Ow"
		ship Geocoris "Eek lo Pok"

mission "Hai Building Mountaintop [1b] Return"
	landing
	name "Return to <planet>"
	description `The Hai have another job for you at <destination>.`
	source "Mountaintop"
	destination "Hai-home"
	clearance
	to offer
		has "Hai Building Mountaintop [1a] Construction Escorts: done"
	on offer
		conversation

> 			`On arrival, you see one of the Geocoris deploy a steel ring around itself. Space-based cranes set about removing the three outer sections of the Geocoris and moving them to the steel ring. After they finish with the first, they progressively dismantle the other seven Geocoris you escorted. Within a few hours, there is a makeshift station with eight Geocoris power generators and a sizable amount of storage and habitat space. The construction workers begin unpacking the contents of the Geocoris storage pods, checking construction seals, engaging principle life support systems, and a massive holoprojector.`

Author: So, the use of it as plural was only un-deprecated on wiktionary *after* this was originally written. There remains very few instances of it and there are also 0 results on google's ngram viewer for the use of Geocorises, so we'd decided that the plural of Geocoris was Geocoris. Particularly because the plural spp. referenced in applicable texts appear to use Geocoris. (ala `2. Geocoris spp [plural species].`)

> 			`On arrival, you see one of the Geocoris deploy a steel ring around itself. Space-based cranes set about removing the three outer sections of the Geocoris and moving them to the steel ring. After they finish with the first, they progressively dismantle the other seven Geocoris you escorted. Within a few hours, there is a makeshift station with eight Geocoris power generators and a sizable amount of storage and habitat space. The construction workers begin unpacking the contents of the Geocoris storage pods, checking construction seals, engaging principle life support systems, and a massive holoprojector.`

Reviewer 4: ```suggestion
			`On arrival, you see one of the Geocoris deploy a steel ring around itself. Space-based cranes set about removing the three outer sections of the Geocoris and moving them to the ring. After they finish with the first, they progressively dismantle the other seven Geocoris you escorted. Within a few hours, there is a makeshift station with eight Geocoris power generators and a sizable amount of storage and habitat space. The construction workers begin unpacking the contents of the Geocoris storage pods, checking construction seals, engaging principle life support systems, and erecting a massive holoprojector.`
```
Just adding in @Bardagh 's suggestion and one fix: you need a verb for `holoprojector`, or you could rephrase it so it's grammatically correct. I'd suggest plopping in a verb b/c I think it'd sound smoother than the rephrases I can think of.

> 			`	The Geocoris captains explain that the Hai had conceived this space station option when the Geocoris was first designed, with some station construction principles used to support its massive cargo capacity. With minor modifications it was a made a trivial effort to dismantle eight Geocoris and rapidly assemble a station. They thank you for your service, and inform you that the Hai government needs your help again.`

Author: ```suggestion
			`	The Geocoris captains explain that the Hai had conceived this space station option when the Geocoris was first designed, with some station construction principles used to support its massive cargo capacity. With minor modifications, it was made a trivial effort to dismantle eight Geocoris and rapidly assemble a station. They thank you for your service and inform you that the Hai government needs your help again.`
```
Spotted an additional issue while batching.

	stopover "Frostmark"
	stopover "Heartvalley"
	stopover "Allhome"
	stopover "Dustmaker"
	stopover "Greenbloom"
	stopover "Skyfarm"
	stopover "Farwater"
	stopover "Greenwater"
	destination "Mountaintop"
	clearance
	passengers 1
	blocked `Terry contacts you and says, "Captain, you're going to need a bunk free for me. Do you have space?" You do not, so you should complete or abort another mission to make room for her, then return here.`
	on visit
		dialog `You have arrived at the Mountaintop construction site but it isn't finished or staffed yet.`
	to offer
		has "Hai Building Mountaintop [1b] Return: done"
	on offer
		conversation
			`At Hai-home you are directed to a now familiar government hangar.`
			`	Sayari, Xilin, and Alondo all send you greeting pings, but are busy at the time of your landing. To your surprise though, Terry is banging on your airlock before too long.`
			`	"Captain <last>, or should I call you <first>? Anyway, doesn't matter. How would you like to make some money?"`
			`	The smile she gives you is something like you imagine a large cat giving the tasty morsel it just identified as its food.`
			`	"Of course you do, everyone likes to make money," she says.`

> 			`	"The Hai have put out the call for volunteers to staff the new Mountaintop station when it's finished. While you've been off watching early construction works, we've been gauging public interest and well... it's not going to work. In order to sell this they're going to need at least four thousand volunteers by the end of the week to get a good handle on what they can expect to need in terms of facilities, amenities, supplies, and various livability requirements. Nearly twelve thousand by the time the station is actually built. Our current estimates suggest they'll reach a quarter of that."`

Author: Livability is more than habitability. It includes quality of life requirements beyond the habitability basics of food, water, shelter and air.

			`	As she's talking Terry has been laying out charts and graphs, and using a handheld Hai holographic device that she seems to have picked up somewhere to show animated diagrams.`
			choice
				`	"How do you know all this?"`

> 			`	Terry raises an eyebrow at you, "You remember we have a supercomputer that could predict the economic situation across the whole of human habited space up to fifty years in the future to within 1% accuracy right?"`

Reviewer 4: ```suggestion
			`	Terry raises an eyebrow at you. "You remember we have a supercomputer that could predict the economic situation across the whole of human habited space up to fifty years in the future to within 1% accuracy, right?"`
```

			`	When she puts it that way, it does suddenly feel like a bit of a stupid question.`
			# And how did that turn out for you eh? lol
			`	"As it turns out, that same computer is predicting that the Syndicate stands to lose billions of credits, per day, for the duration of any crisis evolving from this which involves the uncontrolled release of the Hai secret; however, we have a plan."`
			choice
				`	"And that is?"`
				`	"What's the deal then?"`

> 			`	Terry crosses her arms and leans back against the bulkhead, "We need a recruitment campaign, and if you haven't guessed, you're the man to do it."`
> 			`	"The Syndicate will offer you a sum of 10,000,000 credits to visit every planet in Hai space and do glossies. I will stage manage, and have promotional material ready inside of 3 hours."`

Reviewer 4: ```suggestion
			`	Terry crosses her arms and leans back against the bulkhead, "We need a recruitment campaign, and if you haven't guessed, you're the man to do it. The Syndicate will offer you a sum of 10,000,000 credits to visit every planet in Hai space and do glossies. I will stage manage, and have promotional material ready inside of 3 hours."`
```
A little iffy on the flow of that last sentence. Grammatically, it's fine, but it just feels somewhat off. I'd suggest rephrasing, but I can't really think of a good way to do it.

Author: How about: "I'll obviously stage manage your image, and will have promotional material ready inside of 3 hours."

Reviewer 4: Sounds good!

			choice
				`	"I don't really have a choice in this matter, do I?"`
			`	Terry's answering grin is best described as "predatory".`
				accept
	on stopover
		conversation

> 			`You have finally visited every world in Hai space and sold the pitch to the people. With any luck enough people will sign up to make it work.`
> 			`	Terry indicates that you may now take her to Mountaintop without further stops, but she has some leftover materials so she'll get you to do some final spruiking if you stop anywhere else along the way.`

Reviewer 4: ```suggestion
			`You have finally visited every world in Hai space and sold the pitch to the people. With any luck, enough will sign up to make it work. Terry indicates that you may now take her to Mountaintop without further stops, but then adds, "I do have some leftover materials though, so I'll get you to do some final spruiking if you stop anywhere else along the way."`
```
[Spruiking](https://www.collinsdictionary.com/us/dictionary/english/spruik#:~:text=(spru%CB%90k),extensively%20or%20elaborately%3B%20spiel%3B%20orate) is slang. I think it would be great in dialogue, but not in narration, so if Terry says it instead it fixes the concern.

	on complete

> 		log `You had to go on a lengthy promotional tour to rustle up volunteers for the new station. Terry took care of all the details and seemed especially keen. Presumably she has some marketing background, which may explain her success within the Syndicate.`

Reviewer 4: ```suggestion
		log `Had to go on a lengthy promotional tour to rustle up volunteers for the new station. Terry took care of all the details and seemed especially keen. Presumably she has some marketing background, which may explain her success within the Syndicate.`
```
I think we've had a conversation before about whether log entries are first or second person. I forget what the outcome was, but I do know they're all written from first person in-game at the moment.

		payment 10000000
		conversation

> 			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Shortly after you receive a ping informing you that <payment> has been deposited into your account. Terry must've set it up to happen automatically.`

Reviewer 4: ```suggestion
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Shortly after, you receive a ping informing you that <payment> has been deposited into your account. Terry must've set it up to happen automatically.`
```

				decline

mission "ZZ Hai Building Mountaintop [2a] Sales Job"
	landing
	invisible
	repeat
	source 
		not planet "Hai-home"
		not planet "Mountaintop"
		not planet "Darkwaste"
		government "Hai"
	to offer
		has "Hai Building Mountaintop [2] Recruits: active"
	on offer
		conversation
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop.`
				decline

mission "Hai Building Mountaintop [3] Furnishings"
	name "Retrieve human furnishings."

> 	description "The volunteers are so numerous, that they've worked out that the Hai aren't going to be able to coordinate a realistic access"

Reviewer 5: ```suggestion
	description "The volunteers are so numerous that the Hai need outside help transporting furnishings for their quarters to the new station."
```

	autosave
	landing
	source "Mountaintop"
	stopover "Prime"
	destination "Mountaintop"
	clearance
	on visit

> 		dialog `You have arrived at the Mountaintop construction site but it isn't finished or staffed yet.`

Reviewer 4: ```suggestion
		dialog `You have arrived at the Mountaintop construction site, but it isn't finished or staffed yet.`
```

	to offer
		has "Hai Building Mountaintop [2] Recruits: done"
	cargo "human furnishings" 200

> 	blocked `Sayari contacts you and says, "<first>, you're going to need two hundred tons of cargo space for this." You should return here after freeing up <capacity>.`

Reviewer 4: ```suggestion
	blocked `Sayari contacts you and says, "<first>, you're going to need 200 tons of cargo space for this." You should return here after freeing up <capacity>.`
```

	on offer
		conversation
			`As you and Terry offload her remaining supplies on Mountaintop where she intends to manage the arrivals and setup personally, you get a communication from Sayari come to your personal device.`

> 			`	"Captain <last>! Glad I could get a hold of you on the other side of the wormhole there, I see the new relays are working."`
> 			`	"We've just discovered a problem, interest has soared in sign-ups for Mountaintop, but we've realized that most of our 'human' furnishings produced in Hai space aren't going to pass as having been manufactured with your technology. Mr Yang has pulled some strings, but now we urgently need someone to go pick them up from Prime in Betelgeuse."`
> 			`	"Everything else should be ready on our end soon enough, if you can just handle that pick-up and delivery we'll be ok."`

Reviewer 4: ```suggestion
			`	"Captain <last>! Glad I could get a hold of you on the other side of the wormhole there. I see the new relays are working. We've just discovered a problem: interest has soared in sign-ups for Mountaintop, but we've realized that most of our 'human' furnishings produced in Hai space aren't going to pass as having been manufactured with your technology. Mr Yang has pulled some strings, but now we urgently need someone to go pick them up from Prime in Betelgeuse. Everything else should be ready on our end soon enough, so if you can just handle that pick-up and delivery we'll be okay."`
```

> 			`	You assure her you can handle a simple pick-up.`

Reviewer 4: ```suggestion
			`	You let her know you'll handle it and prepare to leave.`
```
To me, the current wording makes it sound like `<first>` has to convince Sayari they can handle it, but Sayari would be well aware by now that this is definitely within `<first>`'s capability.

				accept
	on stopover

> 		dialog `You have the furnishings the Hai ordered. It's time to return them to <destination>`

Reviewer 5: ```suggestion
		dialog `You have the furnishings the Hai ordered. It's time to return them to <destination>.`
```

	on complete
		event "role-play station active" 12
		event "news: hai video clip debunked"

> 		log `Of all the hilariously mundane things that could trip up a deception, it turned out to be Hai-made furniture. Fortunately you were on hand to save the day by acquiring a large shipment of human-made furniture from Prime in order to complete the station.`

Reviewer 4: ```suggestion
		log `Of all the hilariously mundane things that could trip up a deception, it turned out to be Hai-made furniture. Acquired a large shipment of human-made furniture from Prime in order to complete the station.`
```

		conversation

> 			`By the time you return you are shocked to notice that the station seems to be rapidly approaching a completed state. Some four thousand people have already arrived, though some are still on board ships docked at the station. Once you register your delivery amidst the station chaos it's only a few minutes later that you get a "thank you" message from Sayari. She strongly recommends that you hang around Hai space for the time being while events are ongoing, with a cryptic "P.S. you may get a message from a friend of mine soon."`

Reviewer 4: ```suggestion
			`By the time you return, you are shocked to notice that the station seems to be rapidly approaching a completed state. Some four thousand people have already arrived, though some are still on board ships docked at the station. Once you register your delivery amidst the station chaos, it's only a few minutes later that you get a "thank you" message from Sayari. She strongly recommends that you hang around Hai space for the time being while events are ongoing, with a cryptic "P.S. You may get a message from a friend of mine soon."`
```

event "role-play station active"
	system "Ultima Thule"
		government "Neutral"
		object
			sprite star/f0
			period 10
		object
			sprite planet/desert5
			distance 200
			period 22.3607
		object
			sprite planet/europa-b
			distance 403.41
			period 64.056
		object
			sprite planet/desert7-b
			distance 824.9
			period 187.302
		object
			sprite planet/gas1
			distance 1706.99
			period 557.553
			object
				sprite planet/ice8-b
				distance 216
				period 11.1964
		object "Wormhole Alpha"
			sprite planet/wormhole-syndicate-ad
				"frame rate" 1
				"random start frame"
			distance 4208.63
			period 2158.5
			object Mountaintop
				sprite "planet/station hai eight geocoris"
				distance 315
				period 31.1964
	planet Mountaintop
		government "Neutral"
		attributes station tourism
		landscape land/station5

> 		description `This brand new station is built with wide spacious interiors, though that is largely because it was built for having expansion room inside it to accommodate the Hai's preferred geometric development pattern and long-term perspectives. At the moment there is a number of hai pretending to be humans in costume, and a number of volunteer humans from Hai space helping to make up the act. A powerful holoprojection system is projecting an 'anomaly' into space nearby as a novelty. Some 'investigators' following rumors of the Hai are being entertained at most any time.`

Reviewer 4: ```suggestion
		description `This brand new station is built with wide spacious interiors for later expansion, a staple of Hai building. At the moment, there is a number of Hai pretending to be humans in costume and volunteer humans from Hai space helping to make up the act. A powerful holoprojection system is projecting an 'anomaly' into space nearby as a novelty. It seems to reasonably entertain some 'investigators' following rumors of the Hai.`
```
The explanation for the large interiors felt too academic/explanatory/long winded(?). I think I cut too much out, but I can't seem to come up with anything longer to land it in a nice "middle ground" for length. Also, the `at most any time` just feels really hard to read and understand, so the rephrase is an alternative I came up with.

Author: I'd like to include the geometric construction aspect, since it's only infrequently referred to. other than that I like it though.

Reviewer 4: Edited the original b/c that's easier than making a new suggestion on mobile. How does it sound now?

> 		spaceport `The spaceport is a spacious docking assembly sheltered under broad carapace-like structures which are arrayed around the lower decks like petals. It is designed to look impressive and still has that shiny look and freshly oiled smell of a newly minted construction, though it has only basic facilities and services.`

Reviewer 4: ```suggestion
		spaceport `The spaceport is a spacious docking assembly sheltered under broad, carapace-like structures which are arrayed around the lower decks like petals. It is designed to look impressive and still has that shiny look and freshly oiled smell of a newly minted construction, though it has only basic facilities and services.`
```

# Copyright (c) 2020 MasterOfGrey
#
# Endless Sky is free software: you can redistribute it and/or modify it under the
# terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later version.
#
# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A

> # PARTICULAR PURPOSE.  See the GNU General Public License for more details.

Reviewer 6: ```suggestion
# PARTICULAR PURPOSE. See the GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along with
# this program. If not, see <https://www.gnu.org/licenses/>.
```

		bribe 0

mission "Hai Building Mountaintop [1] Construction Fleet"
	name "Escort construction fleet to Ultima Thule"
	description "Meet a construction fleet at <stopovers> and escort them to to <destination> to prevent a confrontation."
	autosave
	landing
	source "Hai-home"
	stopover "Makerplace"
	destination "Mountaintop"
	clearance
	to offer
		has "Hai Leaks Response 6: done"
	on offer
		conversation
			`When you return to Hai-home, you are directed to a government hangar where you are met by Xilin and Sayari, both looking very relieved.`
			`	"I am very glad to see you returned, Captain <last>. Sayari said you were away on an important errand, but I've been hoping for a day now you would appear. I think you might be the solution to the current impasse."`
			`	"You see," says Sayari, "in the light of the recent civil war your species has had, some of the Elders are less than keen on allowing a partisan observer to see the inner workings of a station in its earliest phases, especially when we haven't built one in so, so long."`
			`	Now that she mentions it, you don't think you've seen a Hai station at all.`
			`	"However, therein lies a problem," says Xilin.`

> 			`	"The Republic will not allow the Hai to build a station this side of the wormhole without some oversight by someone we trust. This has now been a point of contention for some days, and multiple suggestions have been rejected."`

Author: I think more appropriate would be:
```suggestion
			`	"The Republic will not allow the Hai to build a station on our side of the wormhole without some oversight by someone we trust. This has now been a point of contention for some days, and multiple suggestions have been rejected."`
```

			`	You see several appreciative nods happening around the room out of the corner of your eyes.`
			label end
			`	"You may act as the oversight for the human governments in this project. Proposal D-86513-7P4 is approved. Council adjourned."`
			`	When you meet Sayari afterwards, a touch of her usual enthusiasm returns as you share the news.`
			`	"This is fantastic news. Now all that remains to be done is for you to escort the construction fleet from <stopovers> to Ultima Thule and observe the initial stages. They'll be waiting for your arrival."`
				accept
	on accept
		event "role-play station before construction"
		log `The diplomats have elected to double-down on the role-play community angle, citing its success; however, the Hai are unwilling to trust a human government to see the inner workings of a station under construction in the wake of the civil war, and the Republic will not allow a station to be built on their side unsupervised. You have been asked to fulfill the role of supervisor as a result. The Elders asked you some odd questions, but agreed that you were an acceptable individual for the task.`
	on stopover

> 		dialog `A fleet of seven Geocoris are waiting to accompany you when you arrive, with just the final items being loaded as you come into land. They will follow you to the site where the new station will be built.`

Author: When we first did this I spent some time looking it up, 1 source with recent hits used geocorises, and the longer history of its pluralisation just used geocoris as singular and plural. I elected to ignore the outlier even though it was the most recent.

> 		dialog `A fleet of seven Geocoris are waiting to accompany you when you arrive, with just the final items being loaded as you come into land. They will follow you to the site where the new station will be built.`

Reviewer 7: Fair enough. I mainly pointed it out because of the other inconsistencies I noted later; getting them all the same is the biggest thing.

	on complete
		event "role-play station after construction"
	

mission "Hai Building Mountaintop [1a] Construction Escorts"
	landing
	invisible
	source "Makerplace"
	destination "Mountaintop"
	clearance
	to offer
		has "Hai Building Mountaintop [1] Construction Fleet: active"

> 	on visit

Author: But you can't complete 1 without visiting the stopover of Makerplace, and they both complete if you land on Mountaintop after you've been to Makerplace?

> 	on visit

Reviewer 8: But Mountaintop 1a might not complete if you don't have all the escorts with you, in which case Mountaintop 1a triggers its `on visit` while Mountaintop 1 triggers its `on complete`.

> 	on visit

Author: Ohhhhhhhh, surely that could be fixed by requiring the invisible mission to complete first? That way you'd only get the 1a `on visit` text until they were all there, before then getting the 1's `on complete`.

> 	on visit

Reviewer 8: You could stick a `to complete` on the first mission, but I still think it'd be far simpler to just shorted Mountaintop 1 to use the stopover as a destination then have Mountaintop 1a not be invisible.

> 	on visit

Author: Mmmm, I really do like the stopover mechanic though. I think it's criminally underused.

> 	on visit

Reviewer 8: Well you use it like 15 times in the next mission so I think we're covered. :)

> 	on visit

Author: Like true, but it's much more fun suddenly spawning a fleet of followers than just getting some text. XD

			`	Content that the station is built satisfactorily, you prepare to return to <destination>.`
				accept

event "role-play station before construction"
	system "Ultima Thule"
		object
			sprite star/f0
			period 10
		object
			sprite planet/desert5
			distance 200
			period 22.3607
		object
			sprite planet/europa-b
			distance 403.41
			period 64.056
		object
			sprite planet/desert7-b
			distance 824.9
			period 187.302
		object
			sprite planet/gas1
			distance 1706.99
			period 557.553
			object
				sprite planet/ice8-b
				distance 216
				period 11.1964
		object "Wormhole Alpha"
			sprite planet/wormhole
			distance 4208.63
			period 2158.5
			object Mountaintop
				sprite "effect/flotsam yottrite"
				distance 315
				period 31.1964
	planet Mountaintop
		government "Hai"
		attributes uninhabited
		landscape land/station0

> 		description `There's a construction site rapidly assembling a station from Hai Geocoris' orbiting this wormhole.`

Author: Hilariously I'm pretty sure the apostrophe should be used on the one before this, and not on this one.

> 		description `There's a construction site rapidly assembling a station from Hai Geocoris' orbiting this wormhole.`

Author: It would be the captain of that geocoris specifically.

			`	"The Hai have put out the call for volunteers to staff the new Mountaintop station when it's finished. While you've been off watching early construction works, we've been gauging public interest and well... it's not going to work. In order to sell this, they're going to need at least four thousand volunteers by the end of the week to get a good handle on what they can expect to need in terms of facilities, amenities, supplies, and various livability requirements. Nearly twelve thousand by the time the station is actually built. Our current estimates suggest they'll only get a quarter of that unless something changes."`
			`	As she's talking, Terry has been laying out charts and graphs while using a handheld Hai holographic device to show animated diagrams.`
			choice
				`	"How do you know all this?"`
			`	Terry raises an eyebrow at you, "You remember we have a supercomputer that could predict the economic situation across the whole of human habited space up to fifty years in the future to within 1% accuracy right?"`
			`	When she puts it that way, it does suddenly feel like a bit of a stupid question.`

> 			# And how did that turn out for you eh? lol

Author: It was more of an aside than a thing intended to be expanded. This section though was sort of up in the air pending your stated desire to rework the FW stuff.
In this case though, reworking or expanding this conversation is something that's going to happen either with a FW re-work, or the addition of any new alternative campaign. Ideally this section is something I'd like to be "good enough" on the understanding that it'll be frequently revisited in the future with Republic, Syndicate and Pirate campaigns.

> 			`	"As it turns out, that same computer is predicting that the Syndicate stands to lose billions of credits, per day, for the duration of any crisis evolving from this which involves the uncontrolled release of the Hai secret; however, we have a plan."`

Author: Yeah I know there's a push to destroy or otherwise re-purpose the Oracle both by you and some other people. (though as you note I don't know the specific ideas in your head.)

I would really like to see the Oracle explored a little differently though. It's weird that the Oracle is the only AI of it's type in human space, and I'd like to see that explored. My thinking is that any sort of _networked_ AI ultimately goes insane, so they're not used. Given the obvious value of it though, I think it would be fun if (rather than it being networked to be able to draw down all the info it wanted) it was physically isolated to remain sane, and had to be actively fed all the information that made it tick. 
This would then mean that there was a massive infrastructure of corporate data collectors, compilers, and reporters, all feeding back huge volumes of information from across human space to be plugged into the Oracle's databanks so it can do its magic.
You could have a whole department of the Syndicate with a secret purpose, a whole information industry dedicated to making sure that the right information was collected, reported on, and sent to the Oracle for use. I think that has a lot of scope for interesting narrative outcomes - and then the Syndicate could keep their Oracle, with some oversight, because it would be functionally useless without the information industry that feeds it.

			choice
				`	"I don't really have a choice in this matter, do I?"`
			`	Terry's answering grin is best described as "predatory".`
				accept
	on stopover
		"hai dog and pony done" ++
		conversation
			`You have finally visited every world in Hai space and sold the pitch to the people. With any luck, enough will sign up to make it work. Terry indicates that you may now take her to Mountaintop without further stops, but then adds, "I do have some leftover materials though, so I'll get you to do some final spruiking if you stop anywhere else along the way."`
	on complete
		log `Had to go on a lengthy promotional tour to rustle up volunteers for the new station. Terry took care of all the details and seemed especially keen. Presumably she has some marketing background, which may explain her success within the Syndicate.`
		payment 10000000
		conversation

> 			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Shortly after, you receive a ping informing you that <payment> has been deposited into your account. Terry must've set it up to happen automatically.`

Author: Ah this is a legacy item I missed when I reworked the mission to use the dog and pony show.

	on offer
		conversation
			`As you and Terry offload her remaining supplies on Mountaintop where she intends to manage the arrivals and setup personally, you get a communication from Sayari come to your personal device.`
			`	"Captain <last>! Glad I could get a hold of you on the other side of the wormhole there. I see the new relays are working. We've just discovered a problem: interest has soared in sign-ups for Mountaintop, but we've realized that most of our 'human' furnishings produced in Hai space aren't going to pass as having been manufactured with your technology. Mr Yang has pulled some strings, but now we urgently need someone to go pick them up from Prime in Betelgeuse. Everything else should be ready on our end soon enough, so if you can just handle that pick-up and delivery we'll be okay."`
			`	You let her know you'll handle it and prepare to leave.`
				accept
	on stopover
		dialog `You have the furnishings the Hai ordered. It's time to return them to <destination>.`
	on complete
		event "role-play station active" 12

> 		event "news: hai video clip debunked"

Reviewer 8: Really? Where? How?

Author: If this is referring to the event, it's a background event that feeds into News and other things.

Reviewer 8: Seems like the debunking should be something made explicit in the story (although with the movie production idea the debunking would be mentioned as saying it was marketing material).

			`	When she puts it that way, it does suddenly feel like a bit of a stupid question.`
			# And how did that turn out for you eh? lol
			`	"As it turns out, that same computer is predicting that the Syndicate stands to lose billions of credits, per day, for the duration of any crisis evolving from this which involves the uncontrolled release of the Hai secret; however, we have a plan."`
			choice
				`	"And that is?"`
				`	"What's the deal then?"`
			`	Terry crosses her arms and leans back against the bulkhead, "We need a recruitment campaign, and if you haven't guessed, you're the one to do it. The Syndicate will offer you a sum of 10,000,000 credits to visit every planet in Hai space and do glossies. I'll obviously stage manage your image, and will have promotional material ready inside of 3 hours."`
			choice
				`	"I don't really have a choice in this matter, do I?"`
			`	Terry's answering grin is best described as "predatory".`
				accept
	on stopover
		"hai dog and pony done" ++
		conversation

> 			`You have finally visited every world in Hai space and sold the pitch to the people. With any luck, enough will sign up to make it work. Terry indicates that you may now take her to Mountaintop without further stops, but then adds, "I do have some leftover materials though, so I'll get you to do some final spruiking if you stop anywhere else along the way."`

Reviewer 6: The only dictionary where I found `spruiking` referred to it as an archaic Australian slang. I know you want fancy language here, but maybe not _this_ fancy. Also, this might be a bit too informal here.

				defer
			label tedious
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Despite the enthusiasm of the Hai it's tediously repetitive by now.`
				defer
			label tiresome
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. The routine is well-practiced now, but starting to get a little tiresome.`
				defer
			label ordeal
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You struggle to maintain a cheerful and enthusiastic facade as you count down the minutes until this ordeal is over.`
				defer
			label interminable
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				defer
			label toomany
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				defer
			label overdone

> 			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Terry makes good on her promise of forcing you to shuffle out and do some last-minute spruiking to offload the remaining promotional materials to those who missed it the first time.`

Reviewer 6: Same about spruiking.

				decline

mission "Hai Building Mountaintop [3] Furnishings"
	name "Retrieve human furnishings."
	description "The volunteers are so numerous that the Hai need outside help transporting furnishings for their quarters to the new station."
	autosave
	landing
	source "Mountaintop"
	stopover "Prime"
	destination "Mountaintop"
	clearance
	on visit
		dialog `You have arrived at the Mountaintop construction site, but it isn't finished or staffed yet.`
	to offer
		has "Hai Building Mountaintop [2] Recruits: done"
	cargo "human furnishings" 200
	blocked `Sayari contacts you and says, "<first>, you're going to need 200 tons of cargo space for this." You should return here after freeing up <capacity>.`
	on offer
		conversation

> 			`As you and Terry offload her remaining supplies on Mountaintop where she intends to manage the arrivals and setup personally, you get a communication from Sayari come to your personal device.`

Reviewer 6: ```suggestion
			`As you and Terry offload her remaining supplies on Mountaintop where she intends to manage the arrivals and setup personally, you get a communication from Sayari on your personal device.`
```

# terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later version.
#
# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
# PARTICULAR PURPOSE.  See the GNU General Public License for more details.

########################################################################

event "secret mountaintop"
	# Ensure Mountaintop exists so the next mission can trigger
	system "Ultima Thule"
		government "Uninhabited"
		object
			sprite star/f0
			period 10
		object
			sprite planet/desert5
			distance 200
			period 22.3607
		object
			sprite planet/europa-b
			distance 403.41
			period 64.056
		object
			sprite planet/desert7-b
			distance 824.9
			period 187.302
		object
			sprite planet/gas1
			distance 1706.99
			period 557.553
			object
				sprite planet/ice8-b
				distance 216
				period 11.1964
		object "Wormhole Alpha"
			sprite planet/wormhole
			distance 4208.63
			period 2158.5
			object Mountaintop
				sprite "effect/flotsam yottrite"
				distance 315

> 				period 31.1964

Author: Yeah, so we tried that, and in theory a bunch of things are possible with named objects without redefining the whole system.

In practice, those things don't work. So unless someone wants to dive into the code of how that works and either a) fix the code or b) fix the wiki (I'm honestly not sure which thing is wrong), then this is how it's play tested working successfully.

	planet Mountaintop
		government "Unknown"
		# Placeholder description; you can't actually land.
		description `This tiny asteroid has a space hazard warning transmitter, of a type typically used to mark construction sites. Perhaps someone intends to build a space station here?`
		"required reputation" 1000
		bribe 0

mission "Hai Building Mountaintop [1] Construction Fleet"
	name "Escort construction fleet to Ultima Thule"
	description "Meet a construction fleet at <stopovers> and escort them to to <destination> to prevent a confrontation."
	autosave
	landing
	source "Hai-home"
	stopover "Makerplace"
	destination "Mountaintop"
	clearance
	to offer
		has "Hai Leaks Response 6: done"
	on offer
		conversation
			`When you return to Hai-home, you are directed to a government hangar where you are met by Xilin and Sayari, both looking very relieved.`
			branch buildingdeferred

> 				"Building Mountaintop Deferred" > 0

Author: I'm not sure that works with how it increments though? Also, would probably like to, at some point, put in extra text options if you've deferred it multiple times.

> 				"Building Mountaintop Deferred" > 0

Reviewer 9: > 0 is the same as "has"

	destination "Hai-home"
	clearance
	to offer
		has "Hai Building Mountaintop [1a] Construction Escorts: done"
	on offer
		conversation
			`On arrival, you see one of the Geocoris deploy a steel ring around itself. Space-based cranes set about removing the three outer sections of the Geocoris and moving them to the ring. After they finish with the first, they progressively dismantle the other seven Geocoris you escorted. Within a few hours, there is a makeshift station with eight Geocoris' power generators and a sizable amount of storage and habitat space. The construction workers begin unpacking the contents of the Geocoris' storage pods, checking construction seals, engaging principle life support systems, and erecting a massive holoprojector.`
			`	The Geocoris' captains explain that the Hai had conceived this space station option when the Geocoris was first designed, with some station construction principles used to support its massive cargo capacity. With minor modifications, it was made a trivial effort to dismantle eight Geocoris and rapidly assemble a station. They thank you for your service and inform you that the Hai government needs your help again.`

> 			`	Content that the station is built satisfactorily, you prepare to return to <destination>.`

Author: I swear there was a shipyard description or something which described the Geocoris as having been built using some principles from space station construction. Now I can't find it though.

	stopover "Heartvalley"
	stopover "Allhome"
	stopover "Dustmaker"
	stopover "Greenbloom"
	stopover "Skyfarm"
	stopover "Farwater"
	stopover "Greenwater"
	destination "Mountaintop"
	clearance
	passengers 1
	blocked `Terry contacts you and says, "Captain, you're going to need a bunk free for me. Do you have space?" You do not, so you should complete or abort another mission to make room for her, then return here.`
	on visit
		dialog `You have arrived at the Mountaintop construction site, but it isn't finished or staffed yet.`
	to offer
		has "Hai Building Mountaintop [1b] Return: done"
	on offer
		conversation
			`At Hai-home, you are directed to a now familiar government hangar. Sayari, Xilin, and Alondo all send you greeting pings, but are busy at the time of your landing. To your surprise though, Terry is banging on your airlock before too long.`
			`	"Captain <last>, or should I call you <first>? Anyway, doesn't matter. How would you like to make some money?" The smile she gives you is something like you imagine a large cat giving the tasty morsel it just identified as its food. "Of course you do, everyone likes to make money," she continues, not waiting for you to reply.`

> 			`	"The Hai have put out the call for volunteers to staff the new Mountaintop station when it's finished. While you've been off watching early construction works, we've been gauging public interest and well... it's not going to work. In order to sell this, they're going to need at least four thousand volunteers by the end of the week to get a good handle on what they can expect to need in terms of facilities, amenities, supplies, and various livability requirements. Nearly twelve thousand by the time the station is actually built. Our current estimates suggest they'll only get a quarter of that unless something changes."`

Reviewer 10: Do these 4000 volunteers need to be there? If they've only got 1000 so far, and we need to go everywhere, I don't see how it's reasonably possible to achieve this. How long does it take to go straight from Hai-Home to Ultima Thule by hyperdrive?

Author: Changed it to month

			`	As she's talking, Terry has been laying out charts and graphs while using a handheld Hai holographic device to show animated diagrams.`
			choice
				`	"How do you know all this?"`
			`	Terry raises an eyebrow at you, "You remember we have a supercomputer that could predict the economic situation across the whole of human habited space up to fifty years in the future to within 1% accuracy right?"`
			`	When she puts it that way, it does suddenly feel like a bit of a stupid question.`
			# And how did that turn out for you eh? lol
			`	"As it turns out, that same computer is predicting that the Syndicate stands to lose billions of credits, per day, for the duration of any crisis evolving from this which involves the uncontrolled release of the Hai secret; however, we have a plan."`
			choice
				`	"And that is?"`
				`	"What's the deal then?"`

> 			`	Terry crosses her arms and leans back against the bulkhead, "We need a recruitment campaign, and if you haven't guessed, you're the man to do it. The Syndicate will offer you a sum of 10,000,000 credits to visit every planet in Hai space and do glossies. I'll obviously stage manage your image, and will have promotional material ready inside of 3 hours."`

Author: ```suggestion
			`	Terry crosses her arms and leans back against the bulkhead, "We need a recruitment campaign, and if you haven't guessed, you're the one to do it. The Syndicate will offer you a sum of 10,000,000 credits to visit every planet in Hai space and do glossies. I'll obviously stage manage your image, and will have promotional material ready inside of 3 hours."`
```

				defer
			label interminable
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Why did you ever agree to this interminable dog and pony show? Is it really worth all this just to save one tiny little galaxy?`
				defer
			label toomany
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. You're pretty sure you've done this more times than there are habitable planets in Hai space. Did you lose track and repeat a world? Will this nightmare ever end?`
				defer
			label overdone
			`You land on the world of <planet> and do your part to sell the opportunity on Mountaintop. Terry makes good on her promise of forcing you to shuffle out and do some last-minute spruiking to offload the remaining promotional materials to those who missed it the first time.`
				decline

mission "Hai Building Mountaintop [3] Furnishings"
	name "Retrieve human furnishings."
	description "The volunteers are so numerous that the Hai need outside help transporting furnishings for their quarters to the new station."
	autosave
	landing
	source "Mountaintop"
	stopover "Prime"
	destination "Mountaintop"
	clearance
	on visit

> 		dialog `You have arrived at the Mountaintop construction site, but it isn't finished or staffed yet.`

Reviewer 10: This doesn't seem like a useful stopover message.

Author: You only get this if you try and skip stops in Hai space.

	to offer
		has "Hai Building Mountaintop [2] Recruits: done"
	cargo "human furnishings" 200
	blocked `Sayari contacts you and says, "<first>, you're going to need 200 tons of cargo space for this." You should return here after freeing up <capacity>.`
	on offer
		conversation

> 			`As you and Terry offload her remaining supplies on Mountaintop where she intends to manage the arrivals and setup personally, you get a communication from Sayari come to your personal device.`

Reviewer 10: This does not make sense if this mission was blocked because I don't have enough cargo space. Why wouldn't I leave Terry here while I go acquire more cargo space?
